enterprise-suite/scripts/lbc.py

```diff
@@ -284,13 +293,91 @@ def check_resource_list(cmd, expected, fail_msg):
         return all_found
     return False
 
-# Checks for console PVCs
-def are_pvcs_created(namespace):
-    return check_resource_list(
-        cmd='kubectl get pvc --namespace={} --no-headers'.format(namespace),
-        expected=CONSOLE_PVCS,
-        fail_msg='Found some PVCs ({}) from previous console install, but not all expected: {}.\nTo avoid data loss, please remove them manually'
-    )
+# Returns two lists.  For each of the PVC types we care about, the associated PV
+# is in the first or second list if the reclaim policy is RETAIN or not respectively.
+# Each list element is a tuple of PV name and claim.
+def pvsRetainedOrNot(namespace):
+    returncode, stdout, _ = run('kubectl --namespace {} get pv --no-headers'
+                             .format(namespace))
+    retained = []
+    notRetained = []
+
+    if returncode != 0:
+        printerr("Unable to retrieve PV info.  Proceed with caution.")
+    else:
+        for pvc in CONSOLE_PVCS:
+            # We might want to do this by parsing yaml instead...
+            # pvc-accccc27-2ef8-11e9-a408-080027c68b7b   32Gi       RWO            Delete           Bound    lightbend/alertmanager-storage   standard                24h
+            claim = namespace + '/' + pvc
+
+            match = re.search(r'^([\w-]+)\s+\w+\s+\w+\s+(\w+)\s+Bound\s+('+claim+r')\s', stdout, re.MULTILINE)
+
+            if match:
+                if match.group(2) == 'Retain':
+                    list = retained
+                else:
+                    list = notRetained
+                list.append((match.group(1), match.group(3)))
+
+    return (retained, notRetained)
+
+# This is the behavior as described at
+# https://github.com/lightbend/es-backend/issues/572
+def checkPVthings(aboutToUninstall=False, namespace=None):
+    returncode, stdout, stderr = run('helm get ' + args.helm_name,
+                            DEFAULT_TIMEOUT, show_stderr=True)
+
+    if (returncode != 0) and not ('Error: release: "{}" not found'.format(args.helm_name) in stderr):
+        printerr("Unable to retrieve PV info.  Proceed with caution.")
+        return
+
+    hasPVs = 'usePersistentVolumes: true' in stdout
+
+    # This assumes the default is true.  What happens if they modify this in values.yaml?  We'll miss that I think.
+    wantsPVs = len(filter(lambda x: 'usePersistentVolumes=false' in x, sys.argv)) == 0
+
+    # debug.  remove
+    printerr("hasPVs: {}, wantsPVs: {}".format(hasPVs, wantsPVs))
+
+    if ((not hasPVs and wantsPVs) or (not hasPVs and aboutToUninstall)):
+        # This case would be typical for a dev/demo.  Chances are we're okay losing the data.
+        #    (Not sure how useful this is really.  Don't think they can (easily) grab the data.)
+        printerr("WARNING: usePersistentVolumes was false, now true. Continued (un)installation will result in the loss of Console data.")
```

Comment:
This message is confusing in case `hasPVs=false` and `aboutToUninstall=true`, because `usePersistentVolumes` is still not true as far as I can tell?

Reply:
When `hasPVs` is false, it means the data has been stored on non-permanent volumes.  (e.g. `emptyDir`)  If the user does a delete, then they'll lose that data.  Chances are they don't care about it, but I chose not to ignore that case.

Reply:
Sure, that needs a warning message. The part that's confusing me is `usePersistentVolumes was false, now true`, because in case of uninstall (`aboutToUninstall=true`) it is not true.

Reply:
I don't agree with this. We don't need to warn the user if they are using emptyDir. Their data is already ephemeral! If the pod is bounced for some reason, they will lose their data as well. There is no way for them to save their data, so warnings are pointless.

Reply:
@mitkus: I understand what you're saying now.  Good point.  I'll fix if this whole case (warning on loss of ephemeral data) doesn't disappear...

@jsravn: I chose to call out the case of deleting ephemeral data because it might still catch someone off-guard.  (Say you've been running the console all day while working on some prometheus bug.)  I can be convinced otherwise on that though.

Reply:
This is going to be a huge pain though - even in dev mode with emptyDir we'll need to pass in `nowarn`? I think it's taking things too far. We only need to worry about users actually wanting to persist their data with `--usePersistentVolumes`.